### DISCUSSION TITLE
Can not set -server.http.listen-addr in Docker on Linux

### DISCUSSION THREAD

**Discussion by DecipherOne** [grafana-cold-storage/agent#6970] [Q&A] [answered]
Hello Everyone,

I am new to grafana agents and am attempting to setup alloy. Originally I thought it was an issue with the reverse proxy, but everything there seems to be working correctly. Instead, when I try to follow along with instructions where files should be located for updating environment variables via linux, those files do not exist in the container.

When attempting to manually via cmd line or via my docker-compose use the run command it results in the following:

` alloy run
       --server.http.listen-addr=0.0.0.0:12345
       --storage.path=/var/lib/alloy/data
Error: accepts 1 arg(s), received 0`

I have not been able to determine the cause of this , and after about a day of trying to figure this out, am leaning towards abandoning this implementation. If someone can point me in the right direction on how to resolve this, it would be great.

Bypassing my reverse proxy and running this directly in docker, produces the same result.

Thank you 

Alloy Version 1.2.0 Latest

**Comment by rfratto** [comment]
Hi, `alloy run` requires an argument for the location of the config file, which is missing from the command you shared. 

```
alloy run --server.http.listen-addr=0.0.0.0:12345 --storage.path=/var/lib/alloy/data PATH_TO_CONFIG.alloy 
```

**Reply by DecipherOne** [reply]
Ok, so that is automatically supplied when not setting the value manually through the compose file. But if you are setting it manually, you are required to supply it?

It looks like the container initialization is already handling that via docker inspect.

'     "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "TZ=US/Central",
                "PUID=1000",
                "PGID=140",
                "ALLOY_SERVER_HTTP_LISTEN_ADDR=0.0.0.0:3012",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "ALLOY_DEPLOY_MODE=docker"
            ],
            "Cmd": [
                "run",
                "/etc/alloy/config.alloy",
                "--storage.path=/var/lib/alloy/data"
            ],'

**Reply by rfratto** [reply, marked as answer]
Based on the error the argument for the config file isn't being passed; I'm not sure exactly how you're running Alloy in compose but it does seem like you need to add an arugment for the config file.

**Reply by DecipherOne** [reply]
Thank you, that wasn't clear and was causing me all sorts of headaches! Huzzah!

**Reply by DecipherOne** [reply]
For anyone else who stumbles upon this, the syntax for the correct command
'command:
      - run
      - /etc/alloy/config.alloy
      - --server.http.listen-addr=0.0.0.0:12345
      - --storage.path=/var/lib/alloy/data'

**Reply by nikita-toffee-ai** [reply]
Is there any way to set it through the `*.alloy` config file?

**Reply by szvieira** [reply]
> For anyone else who stumbles upon this, the syntax for the correct command 'command: - run - /etc/alloy/config.alloy - --server.http.listen-addr=0.0.0.0:12345 - --storage.path=/var/lib/alloy/data'

Nice bro it worked!!
